lms/djangoapps/discussion/rest_api/views.py

```diff
@@ -154,6 +161,137 @@ def get(self, request, course_id):
         return Response(response)
 
 
+class UserCommentsView(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**:
+
+        Retrieve the list of the user's comments in the specified course.
+
+    **Example Requests**:
+
+        GET /api/discussion/v1/users/{username}/comments/{course_id}
+        GET /api/discussion/v1/users/{username}/comments/{course_id}&flagged=True
```

Comment:
I'm not happy with these paths, I think they deviate a bit too much from the standard paths. I think the ideal situation is that this data is just returned by the current comments API but with new filters. 

i.e. just add a username filter there, if a username is provided, use the rest of the code from here.

Reply:
I've finished merging the endpoint with the existing comments endpoint, including handling the endorsement related serializer methods and documentation, and adjusted the tests accordingly. I'll wait for the CI to run and remove the draft status.